lib/ansible/modules/cloud/amazon/ec2_instance_attribute_facts.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright, (c) 2018, Ansible Project
+# Copyright, (c) 2018, Moshe Immerman
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: ec2_instance_attribute_facts
+short_description: Get EC2 instance attribute facts
+description:
+    -  Get EC2 instance attribute facts
+version_added: 2.8
+author:
+    - Moshe Immerman <moshe.immerman@gmail.com>
+requirements: [ "boto3", "botocore" ]
+options:
+   instance_id:
+        description:
+            - EC2 Instance Id.
+        required: True
+   region:
+         description:
+            - AWS Region of the instance.
+         required: True
+   attribute:
+        description:
+            - The name of the instance attribute. See https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeInstanceAttribute.html
+        required: True
+extends_documentation_fragment:
+    - aws
+    - ec2
+'''
+
+EXAMPLES = """
+# Note: These examples do not set authentication details, see the AWS Guide for details.
+- ec2_instance_attribute_facts:
+    instance_id: "{{ec2_instance.id}}"
+    region: "{{region}}"
+    attribute: "userData"
+  register: userDataFact
+- debug: msg="{{userDataFact.attributes.UserData.Value}}"
```

Comment:
I missed this before, but returned values should be snake_case. See https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/GUIDELINES.md#returning-values

It might also make sense to accept snake_case in the 'attribute' argument and convert it before calling the API.